### PR TITLE
Created "shell" code for Dashboard UI 

### PR DESCRIPTION
**Pull request by AlvinDevelopments**
There was a file I deleted inside the project but the name of was still inside the make file. I just fixed it up so it should work!

### CODE REVIEW

**Comment by enochtsang** [comment]
So the build is still failing, I tried building myself as well and I get the same error.

`ui_widedash.h: No such file or directory`

**Comment by enochtsang** [comment]
So one thing I would like to point out is our style guide, I don't believe I mentioned it before so my apologies. 

Here's a link
https://github.com/UCSolarCarTeam/Development-Guide/tree/master/Style-Guide

Generally, try to follow the style currently in the project, it's pretty obvious which changes are yours right now xD

**Comment by enochtsang** [comment]
You currently have both a DisplayView and a DashboardView0, a ui should only be passed into one view. The `DisplayView` is currently what is actually displaying the ui right now. By passing it into the DisplayView in the view container in line 43 
```c++
, displayView_(new DisplayView(*DashboardUI_))
```

The ui is being called to show in`DisplayView.cpp`  on line 32 with
```c++
    ui_.show();
``` 

I like what you did with the DashboardView, so we should remove DisplayView from the program and move the `ui_.show()` into `DashboardView`.

File: ViewLayer/DashboardView/DashboardView.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                             VehiclePresenter& vehiclePresenter,
+                             PowerPresenter& powerPresenter,
+                             CommunicationPresenter& communicationPresenter,
+                             FaultsPresenter& faultsPresenter,
+                             MpptPresenter& mpptPresenter,
+                             DashboardUI& ui);
+    ~DashboardView();
+
+private:
+    DashboardUI& ui_;
```
Comment (enochtsang) [inline ViewLayer/DashboardView/DashboardView.h]: The type for this should be the interface, not the actual object. Then we can pass in a mock_DashboardUi.

File: ViewLayer/DashboardUI/I_DashboardUI.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <QLabel>
+#include <QTextEdit>
+
+class I_DashboardUI
```
Comment (enochtsang) [inline ViewLayer/DashboardUI/I_DashboardUI.h]: The UI files were named improperly before, all class names should not be camel case, even for abbreviations like UI. So the class name should be `I_DashboardUi`

File: ViewLayer/DashboardUI/DashboardUI.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "I_DashboardUI.h"
+#include "../I_SolarCarWindow/I_SolarCarWindow.h"
+
+namespace Ui {
+    class DashboardUI;
+}
+class DashboardUI : public I_SolarCarWindow, public I_DashboardUI
```
Comment (enochtsang) [inline ViewLayer/DashboardUI/DashboardUI.h]: We probably dont' need the I_SolarCarWindow anymore.

**Comment by bill-luu** [comment]
Is this still open?

**Pull request closed without merging**